src/browser/modules/Main/main.hooks.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2002-2019 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { useState, useEffect } from 'react'
+
+import {
+  PENDING_STATE,
+  CONNECTING_STATE
+} from 'shared/modules/connections/connectionsDuck'
+
+/**
+ * Custom hook for detecting slow connections
+ * @param     {number}      connectionState
+ * @param     {string}      errorMessage
+ * @param     {number}      lastConnectionUpdate
+ * @return    {boolean[]}
+ */
+export function useSlowConnectionState ({
```

Comment:
This looks good, but could you add some tests to demonstrate the expected behavior @huboneo?

Reply:
@oskarhane done!